lib/Frontend/CompilerInvocation.cpp

```diff
@@ -355,6 +355,9 @@ static bool ParseLangArgs(LangOptions &Opts, ArgList &Args,
   if (Args.hasArg(OPT_experimental_dependency_include_intrafile))
     Opts.ExperimentalDependenciesIncludeIntrafileOnes = true;
 
+  if (Args.hasArg(OPT_enable_experimental_differentiable_programming))
```

Comment:
Note: Included this commit from separate PR https://github.com/apple/swift/pull/28001 because it's necessary for the tests to work.